### PR TITLE
Documentation on serializing EC with jackson

### PR DESCRIPTION
**Pull request by yawkat**
This is an old draft blog article of mine. The 'Eclipse Collections' section should probably be removed, explaining that is the job of other docs. Feedback appreciated!

### CODE REVIEW

**Comment by yawkat** [comment]
Will do the ECA tomorrow.

**Comment by donraab** [comment]
Thanks @yawkat! You'll need to sign-off and amend your commit after the ECA. I'll take a look and provide feedback. This is very much appreciated!

File: docs/jackson.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
+# Serializing Eclipse Collections with Jackson
+
+[Eclipse Collections](https://www.eclipse.org/collections/) is a framework that aims to complement the collection library we all know in `java.util`. It offers immutable collections, primitive collections and a handy API to make actually using these types a pleasure.
```
Comment (donraab) [inline docs/jackson.md:3]: Just drop the first paragraph.

File: docs/jackson.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
+# Serializing Eclipse Collections with Jackson
```
Comment (donraab) [inline docs/jackson.md:1]: Include a Copyright Header similar the [ref guide.](https://raw.githubusercontent.com/eclipse/eclipse-collections/master/docs/guide.md)

File: docs/jackson.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+For all the primitive collection types there also exist mutable and immutable versions (`MutableIntList`, `ImmutableIntList`). They behave much the same as their object-based counterparts so we will not cover them here.
+
+Eclipse Collections is split into two parts: The API and the implementation of that API. Since the implementation depends on the API, we will only have to add that as a dependency to our Maven project:
+
+```xml
+<dependency>
+    <groupId>org.eclipse.collections</groupId>
+    <artifactId>eclipse-collections</artifactId>
+    <version>9.2.0</version>
```
Comment (donraab) [inline docs/jackson.md:41]: Update version to 10.4.0

File: docs/jackson.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+</dependency>
+```
+
+Or with Gradle:
+
+```
+compile 'org.eclipse.collections:eclipse-collections:9.2.0'
```
Comment (donraab) [inline docs/jackson.md:48]: Update version to 10.4.0

File: docs/jackson.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+```
+
+9.2.0 is the current Eclipse Collections version at time of writing. Check out [the maven repo](https://search.maven.org/search?q=g:org.eclipse.collections%20AND%20a:eclipse-collections&core=gav) for newer versions.
```
Comment (donraab) [inline docs/jackson.md:51]: I think you can just remove this paragraph.

File: docs/jackson.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+
+Jackson is a library for transforming your data structures from and to JSON and other serialized forms. According to [mvnrepository.com](https://mvnrepository.com/popular) it is the most popular framework for this purpose.
+
+Jackson's architecture consists of three parts: A *core* that is responsible for writing streams of data (typically JSON), *databind* which is responsible for the basic object serialization API, and various Jackson modules that each handle serialization of some special types. Adding Jackson to our project is simple:
+
+```xml
+<dependency>
+  <groupId>com.fasterxml.jackson.core</groupId>
+  <artifactId>jackson-databind</artifactId>
+  <version>2.9.8</version>
```
Comment (donraab) [inline docs/jackson.md:63]: Update to current version.

File: docs/jackson.md [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+</dependency>
+<dependency>
+  <groupId>com.fasterxml.jackson.datatype</groupId>
+  <artifactId>jackson-datatype-eclipse-collections</artifactId>
+  <version>2.9.8</version>
```
Comment (donraab) [inline docs/jackson.md:68]: Update to current version.

File: docs/jackson.md [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+</dependency>
+```
+
+Or with Gradle:
+
+```
+compile 'com.fasterxml.jackson.core:jackson-databind:2.9.8'
+compile 'com.fasterxml.jackson.datatype:jackson-datatype-eclipse-collections:2.9.8'
+```
```
Comment (donraab) [inline docs/jackson.md:77]: Update to current versions.

**Review by donraab — CHANGES_REQUESTED** [review]
Hi @yawkat, thank you very much for the great contribution! I've requested a few changes inline. I think the Eclipse Collections community will really appreciate learning about this.

**Review by donraab — APPROVED** [review]
This looks good to me @yawkat! Please rebase and I will merge.

**Comment by yawkat** [comment]
that checkstyle failure looks like a random error with maven dependency resolution during build.

**Comment by nikhilnanivadekar** [comment]
Yes, it’s an annoying problem. I have restarted the build. Even though the change has nothing to do with checkstyle.

**Pull request merged**